ceno_zkvm/src/scheme/mock_prover.rs

```diff
@@ -1205,6 +1185,84 @@ Hints:
     }
 }
 
+fn compare_lkm<E, K>(
+    lkm_a: LkMultiplicityRaw<K>,
+    lkm_b: LkMultiplicityRaw<K>,
+    errors: &mut Vec<MockProverError<E, K>>,
+) where
+    E: ExtensionField,
+    K: LkMultiplicityKey,
+{
+    let lkm_a = lkm_a.into_finalize_result();
+    let lkm_b = lkm_b.into_finalize_result();
+
+    // Compare each LK Multiplicity.
+    for (rom_type, a_map, b_map) in izip!(ROMType::iter(), &lkm_a, &lkm_b) {
+        if *a_map != *b_map {
```

Comment:
We can simplify the logic here a lot:

```rust
fn compare_lkm<E, K>(
    lkm_a: LkMultiplicityRaw<K>,
    lkm_b: LkMultiplicityRaw<K>,
) -> Vec<MockProverError<E, K>>
where
    E: ExtensionField,
    K: LkMultiplicityKey + Ord,
{
    let lkm_a = lkm_a.into_finalize_result();
    let lkm_b = lkm_b.into_finalize_result();

    // Compare each LK Multiplicity.
    izip!(ROMType::iter(), &lkm_a, &lkm_b)
        .flat_map(|(rom_type, a_map, b_map)| {
            // We use a BTreeSet, instead of a HashSet, to ensure deterministic order.
            let keys: BTreeSet<_> = chain!(a_map.keys(), b_map.keys()).collect();
            keys.into_iter().filter_map(move |key| {
                let count =
                    *a_map.get(key).unwrap_or(&0) as isize - *b_map.get(key).unwrap_or(&0) as isize;

                (count != 0).then_some(MockProverError::LkMultiplicityError {
                    rom_type,
                    key: *key,
                    count,
                })
            })
        })
        .collect()
}
```